src/index.js

```diff
@@ -236,19 +230,63 @@ window.onload = () => {
         ]);
       }
     });
-    whenMapAndDataReady(ddb, map);
   });
+};
+
+// Reload data every five minutes
+const FIVE_MINUTES_IN_MS = 300000;
+const recursiveDataLoad = () => {
+  pageDraws++;
   loadDataOnPage();
+  setTimeout(recursiveDataLoad, FIVE_MINUTES_IN_MS);
+};
 
-  // Reload data every five minutes
-  const FIVE_MINUTES_IN_MS = 300000;
-  const recursiveDataLoad = () => {
-    pageDraws++;
-    loadDataOnPage();
-    setTimeout(recursiveDataLoad, FIVE_MINUTES_IN_MS);
-  };
+// Call only if ddb is updated.
+// Uses a setTimeout to queue a new macrotask
+const callIfUpdated = (callback, delay = 0) => {
+  if (ddb.isUpdated()) {
+    setTimeout(callback, delay);
+  }
+};
 
-  setTimeout(recursiveDataLoad, FIVE_MINUTES_IN_MS);
+document.addEventListener("covid19japan-redraw", () => {
+  callIfUpdated(() => drawKpis(ddb.totals, ddb.totalsDiff));
+  if (!document.body.classList.contains("embed-mode")) {
+    callIfUpdated(() => drawLastUpdated(ddb.lastUpdated, LANG));
+    callIfUpdated(() => drawPageTitleCount(ddb.totals.confirmed));
+    callIfUpdated(() => {
+      trendChart = drawTrendChart(ddb.trend, trendChart);
+    });
+    callIfUpdated(() => {
+      dailyIncreaseChart = drawDailyIncreaseChart(
+        ddb.trend,
+        dailyIncreaseChart
+      );
+    });
+    callIfUpdated(() => {
+      prefectureTrajectoryChart = drawPrefectureTrajectoryChart(
+        ddb.prefectures,
+        prefectureTrajectoryChart,
+        LANG
+      );
+    }, 32);
+    callIfUpdated(() => {
+      prefectureTrendCharts = drawPrefectureTable(
+        ddb.prefectures,
+        ddb.totals,
+        prefectureTrendCharts
+      );
+    }, 32);
+    callIfUpdated(() => drawTravelRestrictions(ddb));
+  }
+
+  callIfUpdated(() => whenMapAndDataReady(ddb, map));
+});
 
+document.addEventListener("DOMContentLoaded", () => {
```

Comment:
Start at DCL instead of Load. Also have tried to make this as simple as possible because its the real entrypoint.